### DISCUSSION TITLE
Uniquify multiple headers of the same name or something?

### DISCUSSION THREAD

**Discussion by BobbyWibowo** [kartikk221/hyper-express#88] [Q&A] [answered]
Hi there,

Suppose I want to have a global middleware that sets a specific Cache-Control header for all requests (thus acting like a default value), but then down the line some more specific routes may set their own Cache-Control headers, how would I go about achieving those routes to only end up with the last set Cache-Control header?

From what I can tell, it was intentional to allow multiple headers of the same name (?)
https://github.com/kartikk221/hyper-express/blob/fb447d1eff76f018db71f71eb8e00f9143d81e6e/src/components/http/Response.js#L172-L173

ExpressJS, at least v4.x, will only keep the last set one, so this is a bit of an unexpected behavior for me

**Comment by kartikk221** [comment, marked as answer]
I'm not sure If I fully understand the expected behavior from your post. If the `header()` call was to use the last set value of a header (essentially replacing/ignoring previous values) then how would multiple middlewares be able to write multiple cookies with the `set-cookie` header during the lifetime of a request?

**Reply by BobbyWibowo** [reply]
From what I can tell, as of Express v4.18.1, it would only allow multiple Set-Cookie headers if set via `res.cookie()` function
Otherwise if set via `res.set()` or `res.header()`, it would only keep the last value, *even if* it is `Set-Cookie` header

Assume the following code snippets:
```js
res.set('Cache-Control', 'no-cache')
res.set('Cache-Control', 'test1')
res.set('Cache-Control', 'test2')
res.set('Cache-Control', 'test3')
res.cookie('name', 'tobi')
res.cookie('rememberme', '1')
```

ExpressJS would end up with the following:
![image](https://i.fiery.me/iPN2.png)

Meanwhile Hyper-Express:
![image](https://i.fiery.me/cYCN.png)

Then suppose down the line there is the following snippet:
```js
res.set('Set-Cookie', 'who is tobi')
```

ExpressJS would then simply end up being:
![image](https://i.fiery.me/uS6D.png)

**Reply by kartikk221** [reply]
Thanks for the breakdown. This issue has been fixed in the recent v6.2.5 release. Hope that helps!

**Reply by BobbyWibowo** [reply]
Awesome! Tyvm!
